### PR TITLE
feat: Stepper interface gains momentum

### PR DESCRIPTION
**Pull request by andiwand**
Stepper interface gains `momentum` functions similar to track parameters

### CODE REVIEW

File: Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp [hunk 1]
```diff
@@ -124,7 +137,7 @@ struct WeightedComponentReducerLoop {
 
 struct MaxMomentumReducerLoop {
   template <typename component_range_t>
-  static const auto& maxMomenutmIt(const component_range_t& cmps) {
+  static const auto& maxAbsoluteMomenutmIt(const component_range_t& cmps) {
```
Comment (felix-russo) [inline Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:140]:
```suggestion
  static const auto& maxAbsoluteMomentumIt(const component_range_t& cmps) {
```

File: Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp [hunk 2]
```diff
@@ -75,7 +76,7 @@ struct WeightedComponentReducerLoop {
   }
 
   template <typename stepper_state_t>
-  static ActsScalar momentum(const stepper_state_t& s) {
+  static ActsScalar absoluteMomentum(const stepper_state_t& s) {
     return std::accumulate(
         s.components.begin(), s.components.end(), ActsScalar{0.},
         [](const auto& sum, const auto& cmp) -> ActsScalar {
```
Comment (felix-russo) [inline Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:82]: I know this was like this before, but maybe it's more readable if the lambda is defined outside of `std::accumulate`?

File: Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp [hunk 3]
```diff
@@ -84,6 +85,18 @@ struct WeightedComponentReducerLoop {
         });
   }
 
+  template <typename stepper_state_t>
+  static Vector3 momentum(const stepper_state_t& s) {
+    return std::accumulate(
+        s.components.begin(), s.components.end(), Vector3::Zero().eval(),
+        [](const auto& sum, const auto& cmp) -> Vector3 {
+          return sum + cmp.weight *
+                           std::abs(cmp.state.absCharge /
```
Comment (felix-russo) [inline Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:94]: Do we need to account for `cmp.state.absCharge = 0`?

Comment (andiwand) [reply Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:94]: charge 0 is kind of broken in all the steppers except the straight line stepper ATM. the future refactor for particle hypothesis is going to improve this hopefully

File: Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp (hunk 2 above)
Comment (andiwand) [reply Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:82]: might be but it is consistent right now and I do not want to intrude the code too much right now

File: Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp [hunk 4]
```diff
@@ -134,40 +147,53 @@ struct MaxMomentumReducerLoop {
...
   static Vector3 direction(const stepper_state_t& s) {
-    return maxMomenutmIt(s.components)
+    return maxAbsoluteMomentumIt(s.components)
         .state.pars.template segment<3>(eFreeDir0);
   }
 
   template <typename stepper_state_t>
-  static ActsScalar momentum(const stepper_state_t& s) {
-    const auto& cmp = maxMomenutmIt(s.components);
+  static ActsScalar qOverP(const stepper_state_t& s) {
```
Comment (paulgessinger) [inline Core/include/Acts/Propagator/MultiEigenStepperLoop.hpp:161]: The diff is a bit funky here, but the two functions `qOverP` and momentum are still separate and both exist.

**Review by paulgessinger — APPROVED** [review]
Looks good I think.

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856287) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856286) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856285) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856284) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsTrkSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856283) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856282) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856281) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856280) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856279) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856278) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856277) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856276) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856275) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856274) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856273) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856272) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30856271) |

**Comment by paulgessinger** [comment]
Run 3 MC failure seems unrelated.

**Pull request merged**